### PR TITLE
Add bot

### PR DESCRIPTION
**Pull request by MariaAfanasyeva**
Added possibility to add bots by users and to update/delete their bots

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import React from "react";
-import BotsList from "./BotsList";
-import Navbar from "./Navbar";
-import Register from "./register";
-import SignIn from "./login";
+import BotsList from "./components/BotsList";
+import Navbar from "./components/Navbar";
```
Comment (artsiom-streetshares) [inline frontend/src/App.js]: why filenames use different case? `UserPage` and `createBot` for example

File: frontend/src/api/api_fetch.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    const token = localStorage.getItem("access_token");
+    const decodedToken = jwt.decode(token);
+    let dateNow = new Date();
+    if (decodedToken.exp < dateNow.getTime() / 1000) {
+      refresh();
+    }
+  }
+
+  const access_token =
+    localStorage.getItem("access_token") !== null
```
Comment (artsiom-streetshares) [inline frontend/src/api/api_fetch.js]: I think you can extract `localStorage.getItem("access_token") `to variable to improve readability

File: frontend/src/api/api_fetch.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      ? "Bearer " + localStorage.getItem("access_token")
+      : null;
+  let options = "";
+  if (login_required === true) {
+    if (method === "GET") {
+      options = {
```
Comment (artsiom-streetshares) [inline frontend/src/api/api_fetch.js]: I think only body is the difference, so why duplicate everything else between if and else blocks

File: frontend/src/api/api_fetch.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      refresh();
+    }
+  }
+
+  const access_token =
+    localStorage.getItem("access_token") !== null
+      ? "Bearer " + localStorage.getItem("access_token")
+      : null;
+  let options = "";
+  if (login_required === true) {
```
Comment (artsiom-streetshares) [inline frontend/src/api/api_fetch.js]: by `login_required` do you mean token required? I don't see a difference between the `if` `and` else blocks. Probably you didn't want to pass the token in the `else`?

File: frontend/src/components/Navbar.js [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+    this.setState({
+      isAuthenticatedUser: value,
+    });
+  };
+
+  componentDidMount() {
+    if (localStorage.getItem("access_token")) {
+      const token = localStorage.getItem("access_token");
+      const decodedToken = jwt.decode(token);
+      const url = `http://127.0.0.1:8000/api/user/${decodedToken.user_id}/info`;
```
Comment (artsiom-streetshares) [inline frontend/src/components/Navbar.js]: let's move this hardcoded host out in the environment in the one of the subsequent PRs

File: frontend/src/App.js (hunk 1 above)
Comment (daniilzinevich) [reply frontend/src/App.js]: All **components** should be CamelCased

File: frontend/src/api/RefreshApi.js [hunk 6]
```diff
@@ -1,6 +1,10 @@
 export const refresh = () => {
```
Comment (daniilzinevich) [inline frontend/src/api/RefreshApi.js]: On the other hand all other files are usually lowerCamelCased

File: frontend/src/auth/Logout.js [hunk 7]
```diff
@@ -4,10 +4,12 @@ class Logout extends Component {
   constructor(props) {
     super(props);
     this.handleClick = this.handleClick.bind(this);
+    console.log(props);
```
Comment (daniilzinevich) [inline frontend/src/auth/Logout.js]: No console logs pls

File: frontend/src/components/CreateBot.js [hunk 8]
```diff
@@ -0,0 +1,275 @@
+import React, { Component } from "react";
+import { api } from "../api/ApiFetch";
+import jwt from "jsonwebtoken";
+
+export default class Create extends Component {
+  constructor(props) {
+    super(props);
+    if (this.props.match.params.bot_id !== undefined) {
```
Comment (daniilzinevich) [inline frontend/src/components/CreateBot.js]: Those only wary in one or two keys. This is not the best way to solve this

**Review by daniilzinevich — APPROVED** [review]

**Pull request merged**